pandas/core/indexes/base.py

```diff
@@ -1130,7 +1130,8 @@ def to_frame(self, index=True):
         """
 
         from pandas import DataFrame
-        result = DataFrame(self._shallow_copy(), columns=[self.name or 0])
```

Comment:
_shallow_copy now does not guarantee that _data is copied, so a test fails. ``to_frame`` should get a copy, else smething like this could happen:

```python
>>> ci = pd.CategoricalIndex(['a', 'b', 'c'])
>>> df = ci.to_frame()
>>> df.iloc[0] = 'c'
>>> df
   0
c  c  # index label changed!
b  b
c  c
```

By using ``self.values.copy()`` rather than ``_shallow_copy``, this issue is avoided.